frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -26,58 +24,66 @@ let eventSourceModels: K8sKind[] = defaultEventSourceModels;
-  const knownSourcesCrd = _.filter(
-    sortModels,
-    (model) =>
-      !!_.find(defaultEventSourceModels, { kind: model?.kind }) &&
-      model?.kind !== EventSourceCamelModel.kind,
-  );
+  const knownSourcesList = [
+    EventSourceApiServerModel.kind,
+    EventSourceContainerModel.kind,
+    EventSourceCronJobModel.kind,
+    EventSourceKafkaModel.kind,
+    EventSourcePingModel.kind,
+    EventSourceSinkBindingModel.kind,
+  ];
+  const knownSourcesCrd = _.filter(sortModels, (model) => knownSourcesList.includes(model.kind));
   const camelSourcesCrd = _.filter(
     sortModels,
     (model) => model?.kind === EventSourceCamelModel.kind,
   );
   const dynamicSourcesCrd = _.filter(
     sortModels,
-    (model) =>
-      !_.find(defaultEventSourceModels, { kind: model?.kind }) &&
-      model?.kind !== EventSourceCamelModel.kind,
+    (model) => !knownSourcesList.includes(model.kind) && model.kind !== EventSourceCamelModel.kind,
   );
   return [...knownSourcesCrd, ...camelSourcesCrd, ...dynamicSourcesCrd];
 };
 
 export const fetchEventSourcesCrd = async () => {
   let eventSourceModelList: K8sKind[] = [];
-  const url = `api/kubernetes/apis/${CustomResourceDefinitionModel.apiGroup}/${
-    CustomResourceDefinitionModel.apiVersion
-  }/${CustomResourceDefinitionModel.plural}?limit=250&labelSelector=${encodeURIComponent(
-    'duck.knative.dev/source=true',
-  )}`;
+  const url = 'api/console/knative-event-sources';
   try {
     const res = await coFetch(url);
     const resolvedRes = await res.json();
-    const allModels = _.map(resolvedRes?.items, (crd) => {
-      const {
-        spec: {
-          group,
-          version,
-          names: { kind, plural, singular },
-        },
-      } = crd;
-      return {
-        apiGroup: group,
-        apiVersion: version,
-        kind,
-        plural,
-        id: singular,
-        label: singular,
-        labelPlural: plural,
-        abbr: kindToAbbr(kind),
-        namespaced: true,
-        crd: true,
-        color: knativeEventingColor.value,
-      };
-    });
-    eventSourceModelList = orderedEventSourceModelData(allModels);
+    const allModels = _.reduce(
+      resolvedRes?.items,
+      (accumulator, crd) => {
+        const {
+          spec: {
+            group,
+            versions,
+            names: { kind, plural, singular },
+          },
+        } = crd;
+        const { name: version } = versions?.find((ver) => ver.served && ver.storage);
+        const modelData = version
+          ? {
+              apiGroup: group,
+              apiVersion: version,
+              kind,
+              plural,
+              id: singular,
+              label: singular,
+              labelPlural: plural,
+              abbr: kindToAbbr(kind),
+              namespaced: true,
+              crd: true,
+              color: knativeEventingColor.value,
+            }
+          : {};
+        return [...accumulator, modelData];
```

Comment:
Its better to just push instead of spreading the array in these scenarios. Its much faster that spreading and does not create a whole new array again every time.

```suggestion
        if (version) {
          accumulator.push({
            apiGroup: group,
            apiVersion: version,
            kind,
            plural,
            id: singular,
            label: singular,
            labelPlural: plural,
            abbr: kindToAbbr(kind),
            namespaced: true,
            crd: true,
            color: knativeEventingColor.value,
          });
        }
        return accumulator;
```

Reply:
The older approach also has a problem where you would add empty objects to the accumulator when there is no version available. So your final array would look some thing like `[model1, model2, {}, {}]`

Reply:
I agree with Rohit. Sorry, I missed this in my review.

Reply:
@rohitkrai03 @sahil143 updated PTAL